server/app/tool/src/management.rs

```diff
@@ -0,0 +1,159 @@
+use crate::management_api::Tee;
+use anyhow::{anyhow, Context, Result};
+use std::fs;
+use std::io::prelude::*;
+use std::path::Path;
+
+use crate::management_api::management_service_client::ManagementServiceClient;
+use crate::management_api::{
+    GetPolicyRequest, GetPolicyResponse, GetReferenceDataRequest, GetReferenceDataResponse,
+    RestoreDefaultPolicyRequest, RestoreDefaultReferenceDataRequest, SetPolicyRequest,
+    SetReferenceDataRequest,
+};
+
+const MANAGEMENT_SOCKET: &str = "127.0.0.1:3001";
+
+impl Tee {
+    fn from_str(tee: &str) -> Result<Tee> {
+        match tee {
+            "sgx" => Ok(Tee::Sgx),
+            "tdx" => Ok(Tee::Tdx),
+            "sev-snp" => Ok(Tee::SevSnp),
+            "sample" => Ok(Tee::Sample),
+            _ => Err(anyhow!("TEE: {} is not supported", tee)),
+        }
+    }
+}
+
+fn store(dir: &Path, name: &str, content: &str) -> Result<()> {
+    let file = dir.join(name);
+    fs::File::create(file.as_path())
+        .context(anyhow!("create file failed"))?
+        .write_all(content.as_bytes())
+        .context(anyhow!("write failed"))
+}
+
+pub async fn set_policy_cmd(
+    tee: Option<&str>,
+    file: Option<&str>,
+    socket: Option<&str>,
+) -> Result<()> {
+    let socket = socket.unwrap_or(MANAGEMENT_SOCKET);
+    if tee.is_none() {
+        return Err(anyhow!("[--tee <TEE>] must be specified."));
+    }
+    if file.is_none() {
+        return Err(anyhow!("[--policy <POLICY>] must be specified."));
+    }
```

Comment:
can this be handled in the command-line parsing part?
and then you don't pass `Option<&str>` to this function.

(same comments for other commands below)